lib/python/qmk/cli/json2c.py

```diff
@@ -17,26 +18,36 @@ def json2c(cli):
 
     This command uses the `qmk.keymap` module to generate a keymap.c from a configurator export. The generated keymap is written to stdout, or to a file if -o is provided.
     """
-    # Error checking
+
+    # Parse the configurator from stdin
```

Comment:
Sadly, Github does not allow comments for blocks with removed lines, so my recommendation is to change [between line 22 and 45](https://github.com/LongerHV/qmk_firmware/blob/json_to_c_stdin/lib/python/qmk/cli/json2c.py#L22-L45) to this:

```python
    try:
        # Parse the json from stdin
        if cli.args.filename.name == '-':
            user_keymap = json.load(sys.stdin)
        # Parse the Configurator json from file
        else:
            if not cli.args.filename.exists():
                cli.log.error('JSON file does not exist!')
                return False
            else:
                user_keymap = json.loads(cli.args.filename.read_text())
    except json.decoder.JSONDecodeError as ex:
        cli.log.error('The JSON input does not appear to be valid.')
        cli.log.error(ex)
        return False
```
This should keep the functionality, while cutting back on code duplication.